src/aggregationserver/aggregationbroker/aggregationreplier.cpp

```diff
@@ -395,7 +442,107 @@ AggServer::AggregationReplier::ProcessWorkloadEventRequest(const AggServer::Work
+
+    std::string start, end;
+
+    // Start time defaults to 0 if not specified
+    if (message.time_interval_size() >= 1) {
+        start = TimeUtil::ToString(message.time_interval()[0]);
+    } else {
+        start = "";
+        //start = TimeUtil::ToString(TimeUtil::SecondsToTimestamp(0));
+    }
+
+    // End time defaults to 0 if not specified
+    if (message.time_interval_size() >= 2) {
+        end = TimeUtil::ToString(message.time_interval()[1]);
+    } else {
+        end = "";
+        //end = TimeUtil::ToString(TimeUtil::SecondsToTimestamp(0));
+    }
+
+    // Get filter conditions
+    const auto conditions = ConditionPairBuilder::createConditionPairs()
+        .add("WorkerInstance.Path", message.worker_instance_paths())
+        .add("Worker.GraphmlID", message.worker_instance_ids())
+        .add("ComponentInstance.Path", message.component_instance_paths())
+        .add("ComponentInstance.GraphmlID", message.component_instance_ids())
+        .add("Component.Name", message.component_names())
+        .finish();
+
+    try {
+        const pqxx::result res = database_->GetMarkerInfo(
+            message.experiment_run_id(),
+            start,
+            end,
+            conditions.getColumns(),
+            conditions.getValues()
+        );
+
+        std::unordered_map<std::string, AggServer::MarkerNameSet*> name_set_map;
+        std::unordered_map<std::string, AggServer::MarkerIDSet*> id_set_map;
+
+        for (const auto& row : res) {
+            // Get label and create new name_set if one doesn't already exist
+            const std::string& label = row["Label"].as<std::string>();
+            AggServer::MarkerNameSet* name_set = nullptr;
+            try {
+                name_set = name_set_map.at(label);
+            } catch (const std::out_of_range& oor_exception) {
+                name_set = response->add_marker_name_sets();
+                name_set->set_name(label);
+                name_set_map.emplace(label, name_set);
+            }
+
+            // Get id and create new id_set if one doesn't already exist for the given name+id combo
+            int id = row["WorkloadID"].as<int>();
+            std::string id_str = label + '_' + std::to_string(id);
+            AggServer::MarkerIDSet* id_set = nullptr;
+            try {
+                id_set = id_set_map.at(id_str);
+            } catch (const std::out_of_range& oor_exception) {
+                id_set = name_set->add_marker_id_set();
+                id_set->set_id(id);
+                id_set_map.emplace(id_str, id_set);
+            }
+
+            // Build MarkerEvent
+            AggServer::MarkerEvent* event = id_set->add_events();
+            auto&& timestamp_str = row["SampleTime"].as<std::string>();
+            bool did_parse = TimeUtil::FromString(timestamp_str, event->mutable_timestamp());
+            if (!did_parse) {
+                throw std::runtime_error("Failed to parse SampleTime field from string: "+timestamp_str);
```

Comment:
I don't think you can use your rvalue timestamp_str after passing into the util function?